src/lib/OS/SimOS.cc

```diff
@@ -169,72 +179,113 @@ void SimOS::createInitialProcess() {
   std::vector<RegisterFileStructure> regFileStructure =
       arch->getRegisterFileStructures();
 
-  if (executablePath_ != DEFAULT_STR) {
+  // Get the tid for new Process
+  uint64_t tid = nextFreeTID_;
+  nextFreeTID_++;
+
+  if (!instructionBytes.empty()) {
+    // Construct Process from `instructionBytes`
+    processes_.emplace(tid, std::make_shared<Process>(
+                                instructionBytes, this, regFileStructure, tid,
+                                tid, sendToMem, memory_->getMemorySize()));
+    // Raise error if created process is not valid
+    if (!processes_[tid]->isValid()) {
+      std::cerr << "[SimEng:SimOS] Could not create process based on "
+                   "supplied instruction span"
+                << std::endl;
+      exit(1);
+    }
+  } else {
+    // Construct Process from specified binary file
+    assert(executablePath_ != DEFAULT_STR &&
+           "[SimEng:SimOS] Tried to construct a Process without specifying a "
+           "pre-compiled binary or raw assembly byte stream.");
     // Concatenate the command line arguments into a single vector and
     // create the process image
     std::vector<std::string> commandLine = {executablePath_};
     commandLine.insert(commandLine.end(), executableArgs_.begin(),
                        executableArgs_.end());
 
-    processes_.emplace(
-        0, std::make_shared<Process>(commandLine, this, regFileStructure, 0, 0,
-                                     sendToMem, memory_->getMemorySize()));
+    // Create new Process. As this is a new process, the TID = TGID.
```

Comment:
Is the TID=TGID also true for creating a process from an insn span?

Reply:
Yes, for any new process the TID will equal the TGID. Only when using `clone` or `fork` is a thread is created where the TGID will be that of the caller and the TID will be globally unique (and so they will differ).

Reply:
Added comment to case when you create a process from an insn span